### PR TITLE
Memory access using libretro API

### PR DESCRIPTION
**Pull request by retro-wertz**
**For Review!**
-use libretro api to access save ram and system ram.
-original mednafen save file handling is intact but commented-out incase we nid to add it back

Please check if disabling the original savefile handling is enough for this, although i did tested with a few games and save and loading savefile looks fine.

Tested using latest RetroArch in Linux-x64

### CODE REVIEW

File: libretro.cpp [hunk 1]
```diff
@@ -862,12 +864,13 @@ static void CloseGame(void)
 {
  if(!BRAMDisabled)
  {
-  std::vector<PtrLengthPair> EvilRams;
+  // Disabled - use libretro api for SaveRAM instead
```
Comment (fr500) [inline libretro.cpp]: Include this in the comment below?
/* */ the whole block instead

File: libretro.cpp [hunk 2]
```diff
@@ -592,13 +593,14 @@ static bool LoadCommon(std::vector<CDIF *> *CDInterfaces)
   memcpy(ExBackupRAM + 0x00, ExBRInit00, sizeof(ExBRInit00));
   memcpy(ExBackupRAM + 0x80, ExBRInit80, sizeof(ExBRInit80));
 
-  FILE *savefp;
+  // Disabled - use libretro API for SaveRAM instead
```
Comment (fr500) [inline libretro.cpp]: Include this in the comment below?
/* */ the whole block instead

File: libretro.cpp [hunk 3]
```diff
@@ -569,9 +571,8 @@ static bool LoadCommon(std::vector<CDIF *> *CDInterfaces)
 
  if(!BRAMDisabled)
  {
-  // Initialize backup RAM
-  memset(BackupRAM, 0, sizeof(BackupRAM));
-  memset(ExBackupRAM, 0, sizeof(ExBackupRAM));
+  // Initialize Save RAM
```
Comment (fr500) [inline libretro.cpp]: not sure it matter in this codebase but I think comments are prefered as /* */ blocks

**Comment by fr500** [comment]
This is great imho, making cores into proper libretro cores instead of relying on the helpers to appear as if they were proper is always good

File: libretro.cpp (hunk 3 above)
Comment (retro-wertz) [reply libretro.cpp]: i think cpp accepts // as comments. will whole block the others though.

**Comment by twinaphex** [comment]
Alright, let's go for it.

**Pull request merged**